src/shared/metadata/cgroup_path_resolver.cc

```diff
@@ -28,11 +28,21 @@
 #include "src/shared/metadata/cgroup_path_resolver.h"
 
 DEFINE_bool(force_cgroup2_mode, true, "Flag to force assume cgroup2 fs for testing purposes");
+DEFINE_bool(force_cgroup1_mode, false, "Flag to force use of cgroup v1 fs");
 
 namespace px {
 namespace md {
 
 StatusOr<std::string> CGroupBasePath(std::string_view sysfs_path) {
+  std::string cgv2_base_path = absl::StrCat(sysfs_path, "/cgroup");
```

Comment:
Unfortunately the OVH case is one where cgroup v1 and v2 are both present, but our logic fails for it.

I agree that this needs to be tested very carefully. After thinking about this more, I think it will be difficult to be confident that all edge cases are covered even if time is invested in testing the known cases.

I'm going to think about this more and see if there is a safer option for handling the OVH case. If the cgroup v1 code doesn't short circuit and allows reaching the cgroup v2 logic, then testing this reordering can be avoided.